components/Common/NavBar.tsx

```diff
@@ -40,15 +41,21 @@ const NavBar: FC = () => {
   const router = useRouter();
   const { events } = useAppSelector((state) => state.events);
   const { blogs } = useAppSelector((state) => state.blogs);
+  const { sponsors } = useAppSelector((state) => state.sponsors);
   const links = [
     { label: 'Events', href: '/Events' },
     { label: 'Blogs', href: '/Blogs' },
+    { label: 'Sponsors', href: '/Sponsors' },
   ];
-  const searchBarWhiteList = ['/Events', '/Blogs'];
+  const searchBarWhiteList = ['/Events', '/Blogs', '/Sponsors', '/Academics'];
   const partialRouteMatch = searchBarWhiteList.some((route) => router.pathname.includes(route));
   const options = [
     ...Object.entries(events).map(([id, { title }]) => ({ label: `Event: ${title}`, value: id })),
     ...Object.entries(blogs).map(([id, { title }]) => ({ label: `Blog: ${title}`, value: id })),
+    ...Object.entries(sponsors).map(([id, { title }]) => ({
+      label: `Sponsors: ${title}`,
+      value: id,
+    })),
```

Comment:
were gonna need to replace store stuff here in the searchbar w react query so it can also show up in the searchbar without redux